### PR TITLE
UCT/IB/MLX5: DONTFORK DEVX buffers fix

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/mlx5/ib_mlx5.h [hunk 1]
```diff
@@ -586,6 +586,8 @@ uct_ib_mlx5_md_buf_alloc(uct_ib_mlx5_md_t *md, size_t size, int silent,
             goto err_free;
         }
         mem->size = size;
+    } else {
+        mem->size = 0;
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.h:590]: set mem->size outside the if

File: src/uct/ib/mlx5/ib_mlx5.h [hunk 2]
```diff
@@ -618,7 +620,7 @@ uct_ib_mlx5_md_buf_free(uct_ib_mlx5_md_t *md, void *buf, uct_ib_mlx5_devx_umem_t
     }
 
     mlx5dv_devx_umem_dereg(mem->mem);
-    if (md->super.fork_init) {
+    if (md->super.fork_init && mem->size) {
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.h:623]: why needed?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5.h:623]: honestly md->super.fork_init won't always be same for alloc and for free

Comment (yosefe) [reply src/uct/ib/mlx5/ib_mlx5.h:623]: how can it be? isn't it decided during md initialization?

Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/ib_mlx5.h:623]: in uct_ib_md_open(), after uct_ib_md_ops->open() call. uct_ib_mlx5_md->zero_mem allocated in uct_ib_mlx5_devx_md_open() when fork_init not set yet

**Pull request merged**